babel.config.js

```diff
@@ -4,7 +4,7 @@ module.exports = (api) => {
     if (isTest) {
         return {
             presets: [
-                ["@babel/preset-react"],
+                ["@babel/preset-react", { runtime: "automatic" }],
```

Comment:
I was testing these build changes locally and I'm getting some errors around the jsx runtime. Please remove these babel/tsconfig changes from this PR. We can add them in a separate PR after discussing more. They should not be needed for this PR. 
```suggestion
                ["@babel/preset-react"],
```

Reply:
yeah this is an interesting one. we should update the presets and react in sync at a later date

Reply:
Hi @gregjopa, please can you explain me a little the problems you're facing because I had the same issue with jsx and runtime error in the console trying to build de project or run the test cases...

Reply:
Also I have a question, I my personal case I prefer *.yml over json to make configuration can I change that. It is not a big deal is just a preference and for me is more readable yaml files.

Reply:
> Also I have a question, I my personal case I prefer *.yml over json to make configuration can I change that. It is not a big deal is just a preference and for me is more readable yaml files.

Let's not change any config files. I'm open to new ideas but I would like to discuss in a new issue and/or offline first.

Please keep this PR scoped to it's original purpose of breaking up the ScriptProvider component into multiple files. The new react-hooks eslint plugin is great so lets keep that too. Adding any new/additional features to this PR makes it difficult for @westeezy and I can review/approve. I would like to get this merged since it's great work and the original problem is solved. Future changes can be separate PRs. It's best practice to keep PR's feature-focused.

Reply:
Yeah, complete agree with you, isn't good for anyone start tracking a bunch of files.